harness/determined/deploy/gcp/gcp.py

```diff
@@ -348,3 +353,57 @@ def set_gcp_credentials_env(tf_vars: Dict) -> None:
 def wait_for_master(configs: Dict, env: Dict, timeout: int = 300) -> None:
     master_url = terraform_output(configs, env, "Web-UI")
     healthcheck.wait_for_master_url(master_url, timeout)
+
+
+def check_or_create_gcsbucket(project_id: str, keypath: Optional[str] = None) -> None:
+    if keypath:
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = keypath
+    bucket_name = project_id + "-determined-deploy"
+    try:
+        storage_service = googleapiclient.discovery.build("storage", "v1")
+    except DefaultCredentialsError:
+        print(
```

Comment:
generally, I think we should avoid `sys.exit` calls in our regular cli code
let's raise a CLIError (for a part of this) similar to https://github.com/determined-ai/determined/pull/7146#discussion_r1231528615

Reply:
done